### DISCUSSION TITLE
Buildroot Root Filesystem to eMMC

### DISCUSSION THREAD

**Discussion by chu551122** [polarfire-soc/polarfire-soc-documentation#184] [Linux build systems for PolarFire SoC] [answered]
Hello, I'm not sure if my understanding is corrector not.
I tried to transfer a Linux cross-compiling example code (i2c) from Ubuntu to Polarfire SoC Icicle Kit in Buildroot.
I used scp to transfer the code and work on Polarfire SoC Icicle Kit well; however, when I power down and power on again, all the ethernet setting and the code I transferred are all disappearred.
Is it due to the Buildroot root file system was in RAM( I saw "buildroot_initramfs_menuconfig" in the makefile)?
How to modify the makefile or menuconfig so that I can set up the Buildroot root file system in eMMC so that I can keep the files I transfer in eMMC to avoid power off and power on lost again.
If my understand is wrong, please correct me and teach me how can I keep the ethernet(static IP) and the file transfer in Buildroot with Polarfire SoC Icicle Kit. Thank you very much!

**Comment by hughbreslin** [comment, marked as answer]
Hi @chu551122 

> Is it due to the Buildroot root file system was in RAM( I saw "buildroot_initramfs_menuconfig" in the makefile)?

Yes this is due to initramfs as none of the data is written back. 

> How to modify the makefile or menuconfig so that I can set up the Buildroot root file system in eMMC so that I can keep the files I transfer in eMMC to avoid power off and power on lost again.

This isn't something we're really supporting at the moment, its a bit complicated with the current buildroot implementation and we'll be moving to a new one shortly which will make this process easier. If you can wait for the new version it would probably be easier to get set up and going with a rootfs. 

You can avoid initramfs using our yocto builds, would you be able to use them?

**Reply by chu551122** [reply]
Hi, Hugh:
    Noted, look forward to new version build procedure release for Buildroot with Polarfire SoC.
    Thank you very much!

**Reply by hughbreslin** [reply]
No problem Michael 🙂

**Reply by chu551122** [reply]
Is there any update for Buildroot to eMMC instead of RAM for configuration? It's a little inconvenient to set the network configuration and transfer application everytime when power on/off.
Thank you very much!

**Reply by hughbreslin** [reply]
No there will be no update to the Buildroot SDK in this org to support this. We are planning to move to buildroot external going forward but have no definitive date yet for the release supporting PolarFire SoC.

**Reply by rockoguido** [reply]
hi.  looking at this thread it seems at the moment the use of Menuconfig with Buildroot will not be supported in the near term for the PolarFire SoC, is that still a valid statement?    
I have a PolarFire SoC customer coming online and they wish to use Buildroot and configuring the system without Menuconfig seems problematic.

**Reply by rockoguido** [reply]
I believe we are going to converge on one build system that targets both PolarFire SoC and Microchip ARM MPUs with the use of Buildroot, that's my understanding.    Is  that the case?

**Reply by chu551122** [reply]
Yes, look forward to see root file system together with the Buildroot system in either eMMC or Other storage system instead of RAM in the near future,
Thank you very much!

**Reply by hughbreslin** [reply]
@rockoguido no the thread is that rootfs isn't available for Buildroot only initramfs. Menu config should work for the Buildroot SDK. We will be merging in with [Microchip Buildroot External](https://github.com/linux4sam/buildroot-external-microchip) and support is officially scheduled for this Friday.

**Comment by rockoguido** [comment]
Hi Hugh, ok guess I misread this thread some.    maybe I should  start another thread about the use of Menuconfig for the PF SoC?    I say this since there are no config files in the configs directory targeting this platform.    Yes I could use Menuconfig to build say for one of Microchip's MPUs but not for the Icicle Kit, at least not with the current delivery.   However from what I read here by the end of this week this will change which is great news!   
If you would like I can open a new discussion on this.  
Thx!

**Reply by hughbreslin** [reply]
What order are you running your commands in? I know its not explicitly mentioned in the Buildroot External readme but you should be able to run:

```
BR2_EXTERNAL=../buildroot-external-microchip/ make icicle_defconfig
make menuconfig
``` 

If you want to make a new thread we can copy over the question and mark this as answer (if it works!)?

**Reply by rockoguido** [reply]
Hi.. I'll keep this thread for now if that's ok   Yes what you write makes sense, the issue is as far as I know I've never seen the defconfig file for the Icicle Kit that you call out.   I just went and looked at the current and some recent Buildroot distributions.   
thx!
dave

**Reply by chu551122** [reply]
![image](https://user-images.githubusercontent.com/55179520/203662391-058c8749-f962-442c-bfcb-d1d3619ba95d.png)
"make buildroot_initramfs_menuconfig DEVKIT=icicle-kit-es"
This is what I learned from Microchip Developer Youtube for buildroot menuconfig setting way. FYR

**Reply by hughbreslin** [reply]
@rockoguido it should be [this one](https://github.com/linux4sam/buildroot-external-microchip/blob/master/configs/icicle_defconfig).

@chu551122 we’re talking about Buildroot external and not the PolarFire SoC Buildroot SDK, the steps in your screenshot are for the PolarFire SoC Buildroot SDK

**Reply by chu551122** [reply]
Hi, Hugh: 
   How to go through BR2_External command for menuconfig as you mentioned? What should I download in advance? And where do I find the complete instruction as I did with Polarfire SoC Buildroot SDK in Github?
   Thank you very much!
![image](https://user-images.githubusercontent.com/55179520/203876773-7dab140c-b537-4490-b22e-4512d446692f.png)
